Android foregound service to allow background syncing

johnny9 opened the pull request:
The idea here is to start a foreground notification service so that the process manager on android won't kill our application when in the background so that we can continue to sync without requiring the device to be awake and our activity in the foreground at all times. 

[![Windows](https://img.shields.io/badge/OS-Windows-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/win64/insecure_win_gui.zip?branch=pull/296)
[![Intel macOS](https://img.shields.io/badge/OS-Intel%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos/insecure_mac_gui.zip?branch=pull/296)
[![Apple Silicon macOS](https://img.shields.io/badge/OS-Apple%20Silicon%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos_arm64/insecure_mac_arm64_gui.zip?branch=pull/296)
[![ARM64 Android](https://img.shields.io/badge/OS-Android-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/insecure_android_apk.zip?branch=pull/296)

johnny9 commented:
Update from[ 3df9234 ](https://github.com/bitcoin-core/gui-qml/commit/3df923434fdf41bcea566abcf4bbd9ad88df44d8)to f62e42f:
  - Simplified the implementation greatly by not relying on qandroidextras module. Instead, the service will just be created as soon as the android Activity starts.

johnny9 commented:
Update  from f62e42f to dc8efb5:
  - Added missing super call in service onStartCommand method
  - Added backwards compatible startService command
  - Rebased with main

jarolrod commented on src/qt/android/src/org/bitcoincore/qt/BitcoinQtService.java:
```diff
@@ -0,0 +1,45 @@
+
+        Notification notification = new Notification.Builder(this, "bitcoin_channel_id")
+            .setSmallIcon(R.drawable.bitcoin)
+            .setContentTitle("Bitcoin")
```
per feedback, It seems we could do without this

johnny9 commented:
Update from[ dc8efb5 ](https://github.com/bitcoin-core/gui-qml/commit/dc8efb52b003b31868580bbea5045863708b0e67)to 06d9871:
  - Rebased with main
  - Removed notification content text

jarolrod commented:
ACK 06d9871428e1c582523f023e554122f09f7a0883

This has been extensively tested with different android devices, and one device synced to tip on main
